client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker-preview.tsx

```diff
@@ -591,38 +592,76 @@ const UnifiedDesignPickerPreview = ( {
 		screens,
 	};
 
+	const closeEligibilityModal = () => {
+		recordTracksEvent( 'calypso_automated_transfer_eligibility_modal_dismiss', {
+			flow: 'onboarding',
+			theme: selectedDesign?.slug,
+		} );
+		setShowEligibility( false );
+	};
+
+	// Check if the theme requires a plan upgrade by comparing sitePlanSlug and requiredPlanSlug.
+	const isPlanSufficient = () => {
```

Comment:
This feels like some logic we could write some unit tests for. Could extract it to a helper function to make that easier.

Reply:
Good idea, added some tests!